app/controllers/students_controller.rb

```diff
@@ -85,6 +110,31 @@ def update_tags
     redirect_to tutor_path(@student)
   end
 
+  def reinstate
+    @student = Student.of(current_user).find(params[:id])
+    @student.update(
+      deleted_on: nil,
+      deleted_by: nil
+    )
+
+    redirect_to @student
+  end
+
+  def delete
+    @student = Student.of(current_user).find(params[:id])
+    unmatch_current_tutor(@student)
+    @student.update(
+      deleted_on: Date.today,
+      deleted_by: current_user.id
+    )
+
+    if current_user.admin?
+      redirect_to @student
+    else
+      redirect_to students_path
+    end
+  end
+
   def destroy
     @student = Student.of(current_user).find(params[:id])
```

Comment:
already done by `:set_student`
i can delete them in my new pr if these are redundant.